### PR TITLE
add all_reduce testcase for 1x8 llmbox configuration

### PR DESCRIPTION
**Pull request by hkwonTT**
Add new testcase for basic all_reduce operation under 1x8 mesh configuration

### Ticket
#2070 
### Problem description
Need testcase for basic all_reduce operation under 1x8 mesh configuration

### What's changed
Add new testcase for basic all_reduce operation under 1x8 mesh configuration.
This testcase runs all_reduce with 'sum' operation across last dimension under 1x8 mesh configuration.
Input/Output tensors have 1x1x128x2048 dimension. 2048 for last dimension came from {tile size(==32) * reduced_scatter shard shape(==8) * all_reduce shard shape(==8) }

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by wooseokTT** [comment]
@hkwonTT  Can you add more test cases that touches various attribute corner cases? It would be great.

**Comment by hkwonTT** [comment]
> @hkwonTT Can you add more test cases that touches various attribute corner cases? It would be great.

Sure. I will add more testcases and open this PR again.

File: test/ttmlir/Silicon/TTNN/llmbox/ccl/ccl_1x8.mlir [hunk 1]
```diff
@@ -14,3 +14,17 @@ func.func @all_gather_cluster1(%arg0: tensor<1x1x32x128xf32>) -> tensor<1x1x32x1
   // CHECK: "ttnn.mesh_shard"
   return %5 : tensor<1x1x32x128xf32>
 }
+
+func.func @forward2(%arg0: tensor<1x1x128x2048xf32>) -> tensor<1x1x128x256xf32> {
```
Comment (tapspatel) [inline test/ttmlir/Silicon/TTNN/llmbox/ccl/ccl_1x8.mlir:18]: nit: rename this to all_reduce_cluster1
